jmclient/jmclient/output.py

```diff
@@ -46,10 +46,13 @@ def fmt_utxo(utxo):
     return utxostr
 
 def fmt_tx_data(tx_data, wallet_service):
-    return 'path: {}, address: {}, value: {}'.format(
-        wallet_service.get_path_repr(wallet_service.script_to_path(tx_data['script'])),
-        wallet_service.script_to_addr(tx_data['script']), tx_data['value'])
-
+    try:
+        path = wallet_service.script_to_path(tx_data['script'])
+        addr = wallet_service.script_to_addr(tx_data['script'])
+        val = tx_data['value']
+    except AssertionError:
```

Comment:
Assertions should not be used for control flow. running with `python -O` (optimized bytecode) will strip out assertions and lead to undefined behaviour in such a case (here probably a different exception somewhere deep down).

Reply:
`script_to_path()` should not fail so I guess the problem is that FB wallets don't handle `script_to_addr()`? Make them return `None` for timelocked scripts maybe?

Reply:
Seeing your comment further down `script_to_path()` is indeed the problem. I see two solutions: (1) call `is_known_script()` before or (2) parse the script if it is a timelocked script and add it to the map.

Reply:
Yeah that seems like a reasonable thing to do, but while writing the test it occurred to me it might make more sense to actually allow an arbitrary script and show it as unrecognized for the podle error report. It seemed logical to me to allow this test to handle any script, including ones that might be created in the future.

The root source of the error is something I commented earlier, that these tests actually load a wallet with old legacy utxos and so the DummyWallet should subclass LegacyWallet, here. If it were a FB wallet you could deal with it differently. I think that could be changed in a future PR to upgrade these tests.

 In real execution you wouldn't want to add an unknown script to the map.

Reply:
Addressed for now in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/969/commits/b73f0a20ab56a20842a9880253abf4dd19d22e61

This removes the assertion checks and just takes the same philosophy as in the other taker tests of 'hack the wallet since we are testing the functionality of the Taker class', by adding a patched `is_standard_wallet_script` method.

As I mentioned somewhere, I forget where, we need to redo the taker_test_data to be all native segwit and possibly add some FB in there, then we can *somewhat* rationalize the tests. I think it can be done later.